src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressDisplayNameTest.cs

```diff
@@ -11,6 +11,7 @@ public class MailAddressDisplayNameTest
         private const string Address = "test@example.com";
         private const string DisplayNameWithUnicode = "DisplayNameWith\u00C9\u00C0\u0106\u0100\u0109\u0105\u00E4Unicode";
         private const string DisplayNameWithNoUnicode = "testDisplayName";
+        private const string DisplayNameWithDoubleQuotes = "test\"Display\"Name";
```

Comment:
Add test cases:
- "\"John Doe\"" ... double quotes at the end
- "Hello \"world hello\" world" ... double quotes in the middle around spaces - typical use case
- "Hello \"world" ... just one double quote (technically kind of incorrect value)

Reply:
More test cases:

- "Hello ""world" ... quotes next to each other
- """" ... only quotes (I don't know if it won't be rejected by `MailAddress` ctor)
- """ ... single quote, nothing else
- "Hello \"world hello\" world" ... escaped backslash before quotes (what's the behavior for this right now?)